File: src/main/java/com/plugin/frege/stubs/FregeMethodStub.java
package com.plugin.frege.stubs;

import com.intellij.psi.stubs.IStubElementType;
import com.intellij.psi.stubs.StubElement;
import com.intellij.util.io.StringRef;
import com.plugin.frege.psi.FregePsiMethod;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

// TODO keep more data
public class FregeMethodStub extends FregeNamedStub<FregePsiMethod> {
    public FregeMethodStub(StubElement parent, @NotNull IStubElementType elementType, @Nullable StringRef name) {
        super(parent, elementType, name);
    }

    public FregeMethodStub(StubElement parent, @NotNull IStubElementType elementType, @Nullable String name) {
        super(parent, elementType, name);
    }
}

File: src/main/kotlin/com/plugin/frege/stubs/FregeMethodStub.kt
package com.plugin.frege.stubs

import com.intellij.psi.stubs.IStubElementType
import com.intellij.psi.stubs.StubElement
import com.intellij.util.io.StringRef
import com.plugin.frege.psi.FregePsiMethod

// TODO keep more data
class FregeMethodStub : FregeNamedStub<FregePsiMethod?> {
    constructor(
        parent: StubElement<*>?,
        elementType: IStubElementType<*, *>,
        name: String?
    ) : super(parent, elementType, name)

    constructor(
        parent: StubElement<*>?,
        elementType: IStubElementType<*, *>,
        name: StringRef?
    ) : super(parent, elementType, name)
}
